### PR TITLE
Copy Inslice() to those parts that use it

### PR DESCRIPTION
**Pull request by ChiuWang**
Signed-off-by: Chao Wang <wangchao.fnst@cn.fujitsu.com>

**- What I did**
Copy Inslice() to those parts that use it.
**- How I did it**
According to @thaJeztah 's suggestion at [/moby/moby/pull/35034(closed)](https://github.com/moby/moby/pull/35034#pullrequestreview-68582683)
Accordance with the tasklist at [Separate the GenerateRandomAlphaOnlyString function from stringutils #35313](https://github.com/moby/moby/pull/35313)
**- A picture of a cute animal (not mandatory but encouraged)**
hahaha

### CODE REVIEW

File: daemon/oci_linux.go
Comment (thaJeztah) [inline daemon/oci_linux.go]: I think we should not make this case-insensitive for this one; volume options are case-sensitive (i.e., `:z` and `:Z` are different options). Doing this comparison case-insensitive looks to be a bug that we had

ping @cpuguy83 wdyt?

File: profiles/seccomp/seccomp.go
Comment (thaJeztah) [inline profiles/seccomp/seccomp.go]: Splitting this utility to the various parts where we use it is probably a good opportunity to check if these should all be case-insensitive as well.

ping @justincormack could you have a look to see if these should all be case-insensitive?

File: integration-cli/docker_api_inspect_test.go
Comment (thaJeztah) [inline integration-cli/docker_api_inspect_test.go]: looks like here it doesn't have to be case-insensitive as well

File: integration-cli/docker_cli_by_digest_test.go
Comment (thaJeztah) [inline integration-cli/docker_cli_by_digest_test.go]: same for this one (no need to be case-insensitive); I suggest changing the `internal/testutil/stringutils.go` to be case-sensitive, as it looks like we only use it to compare exact values.

File: daemon/caps/utils_unix.go
Comment (thaJeztah) [inline daemon/caps/utils_unix.go]: This should be `!inSlice(drops, cap[4:]) {` (you accidentally removed the `!`

**Review by thaJeztah — CHANGES_REQUESTED** [review]
@charrywanganthony tests are failing; I left a comment, which may be related to that failure

Also left some suggestions 👍

File: daemon/caps/utils_unix.go
Comment (charrywanganthony) [reply daemon/caps/utils_unix.go]: Thanks! I'll check it.

File: integration-cli/docker_cli_by_digest_test.go
Comment (charrywanganthony) [reply integration-cli/docker_cli_by_digest_test.go]: Modified. except `inslice()`  in `seccomp.go`, I'm not sure about that. Waiting for the feedback.

File: integration-cli/docker_api_inspect_test.go
Comment (justincormack) [reply integration-cli/docker_api_inspect_test.go]: definitely not.

File: profiles/seccomp/seccomp.go
Comment (justincormack) [reply profiles/seccomp/seccomp.go]: technically this would be an API change if we changed the case sensitivity, although it never needed to be insensitive.

**Comment by charrywanganthony** [comment]
Modified. Except `utils_unix.go`, we changing the `inslice()` utility case-sensitive separately. @thaJeztah @justincormack  @vdemeester  @dnephin PTAL :)

**Review by thaJeztah — APPROVED** [review]
LGTM, but found one nit, which would be nice to have addressed before merging

Thank you!

File: internal/testutil/stringutils_test.go
Comment (charrywanganthony) [reply internal/testutil/stringutils_test.go]: Modified. And a test need rerun.

**Comment by charrywanganthony** [comment]
@vdemeester @dnephin PTAL :monkey:

File: internal/testutil/stringutils.go
Comment (dnephin) [inline internal/testutil/stringutils.go]: I don't think we need our own `testutil` for this. `testify/assert` has `assert.Contains()` which does this already.

The error message we get on failure of `c.Assert(testutil.InSlice(...), checker.Equals, true)` is going to be pretty useless. `Expected true got false` does not help anyone debug a test.

I think we should use `assert.Contains()` instead, which will give a helpful error message. We have already started using these assertions in a few places in the `integration-cli` suite.

**Review by dnephin — COMMENTED** [review]
Thanks for making this change.

Just one more small nit for the test case usage.

File: internal/testutil/stringutils.go
Comment (charrywanganthony) [reply internal/testutil/stringutils.go]: Good idea 👍  Modified. 🐸

Comment (charrywanganthony) [reply internal/testutil/stringutils.go]: and a test need rebuild？

**Review by thaJeztah — APPROVED** [review]
still LGTM

restarted the "experimental" build, because Jenkins had an issue

**Review by vdemeester — APPROVED** [review]
LGTM 🐮

**Comment by apsist025350** [comment]
Line

**Comment by apsist025350** [comment]
Line

**Comment by thaJeztah** [comment]
removed spam comments

**Pull request merged**
